apis/duck/v1alpha1/condition_set.go

```diff
@@ -126,14 +118,23 @@ var _ ConditionManager = (*conditionsImpl)(nil)
 // +k8s:deepcopy-gen=false
 type conditionsImpl struct {
 	ConditionSet
-	accessor ConditionsAccessor
+	status interface{}
 }
 
-// Manage creates a ConditionManager from an object that implements
-// ConditionsAccessopr using the original ConditionSet as a reference.
-func (r ConditionSet) Manage(accessor ConditionsAccessor) ConditionManager {
+// Manage creates a ConditionManager from a status object using the original
+// ConditionSet as a reference. Status must be or point to a struct.
+func (r ConditionSet) Manage(status interface{}) ConditionManager {
+	accessorValue := reflect.Indirect(reflect.ValueOf(status))
+
+	// If status is not a struct, don't even try to use it.
```

Comment:
Should this return `error` instead of a thing that doesn't do things?

Reply:
if we do that then you can't do the `thingy.Manage(anotherthingy).DoSomething()`

Reply:
It seems like the problem here is that the use of reflection introduces the possibility of error (if the supplied type does not match), but we don't want to litter all of our code with error checks for that case (also reasonable).

The current code silently fails if you pass in the wrong `anotherthingy`, particularly in the case of `SetCondition` or the like.

This may be an appropriate use case for `panic` -- we expect that with something like generics (or the ability to declare an interface on field names rather than methods) we could use the type system to enforce this.